lib/java/com/google/android/material/dialog/res/layout/mtrl_alert_select_dialog_multichoice.xml

```diff
@@ -20,12 +20,9 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:minHeight="?attr/listPreferredItemHeightSmall"
-    android:gravity="center_vertical"
+    android:gravity="left|center_vertical"
     android:paddingLeft="@dimen/abc_select_dialog_padding_start_material"
     android:paddingRight="?attr/dialogPreferredPadding"
-    android:paddingStart="@dimen/abc_select_dialog_padding_start_material"
-    android:paddingEnd="?attr/dialogPreferredPadding"
     android:drawableLeft="?android:attr/listChoiceIndicatorMultiple"
-    android:drawableStart="?android:attr/listChoiceIndicatorMultiple"
     android:drawablePadding="20dp"
     android:ellipsize="marquee" />
```

Comment:
I've been looking into this to see if there's a way to keep one, non-qualified, resource file instead of splitting out a v17 specific layout. It looks like you can fix the issue this PR addresses by changing the original multichoice and singlechoice layouts to:

```
  <!--  A left android:gravity will handle text alignemnt on API 16 and below while-->
  <!--  android:textAlignemnt will be used on 17+. A start android:gravity is not added -->
  <!--  to avoid text being aligned right while a drawable is aligned left in RTL configurations -->
  <!--  on API 16 and below.-->

  <CheckedTextView
     ...
     android:gravity="left|center_vertical" 
     android:textAlignment="viewStart"
     ...
     app:drawableLeftCompat="?android:attr/listChoiceIndicator(Single|Multiple)"
     app:drawableStartCompat="?android:attr/listChoiceIndicator(Single|Multiple)"
     .../>
```

As far as I've tested, this produces the same result as the changes you've provided for API 16, 17 and 17+. It would be great to avoid splitting out the layouts if possible!